include/onnxruntime/core/session/onnxruntime_ep_c_api.h

```diff
@@ -257,6 +263,34 @@ struct OrtEp {
   OrtStatus*(ORT_API_CALL* GetPreferredDataLayout)(_In_ OrtEp* this_ptr,
                                                    _Out_ OrtEpDataLayout* preferred_data_layout);
 
+  /** \brief Given an op with domain `domain` and type `op_type`, determine whether an associated node's data layout
+   *         should be converted to `target_data_layout`.
+   *         If the EP prefers a non-default data layout (see `GetPreferredDataLayout()`), this function will be called
+   *         during layout transformation with `target_data_layout` set to the EP's preferred data layout.
+   *
+   * \note Implementation of this function is optional.
+   *       If an EP prefers a non-default data layout, it may implement this to customize the specific op data layout
+   *       preferences at a finer granularity.
+   *
```

Comment:
I'm not sure if we have official documentation regarding the ORT layout transformer, but maybe we could link to these dev notes that explain the relationship between `GetPreferredDataLayout`, `GetCapability`, and "layout-sensitive ops": https://github.com/microsoft/onnxruntime/blob/main/onnxruntime/core/optimizer/layout_transformation/layout_transformation_dev_notes.md

Reply:
AFAIK there's no existing documentation as it was always an internal feature (hence not part of the previous EP API).